inference-engine/src/mkldnn_plugin/nodes/mkldnn_depth_to_space_node.cpp

```diff
@@ -0,0 +1,351 @@
+    auto &srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
+    if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
+        THROW_ERROR << "has not allocated destination memory";
+    if (!srcMemPtr || !srcMemPtr->GetPrimitivePtr())
+        THROW_ERROR << "has not allocated input memory";
+    if (getSelectedPrimitiveDescriptor() == nullptr)
+        THROW_ERROR << "has unidentified preferable primitive descriptor";
+
+    SizeVector srcDims = getParentEdgeAt(0)->getBlob()->getTensorDesc().getDims();
+    SizeVector dstDims = getChildEdgeAt(0)->getBlob()->getTensorDesc().getDims();
+    prepareParams(srcDims, dstDims);
+
+    size_t nDims = srcDims.size();
+    const size_t nSpatialDims = nDims - 2;
+    const bool isBlocked = getParentEdgeAt(0)->getMemory().GetDesc().isBlockedCFormat();
+    nDims += nSpatialDims + static_cast<int>(isBlocked) + static_cast<int>(isBlocked && mode == Mode::DEPTH_FIRST);
+    const size_t lastIdx = nDims - 1;
+
+    order.resize(nDims);
+    optimizedParams.src_block_dims.resize(nDims);
+    order[0] = 0;
+    optimizedParams.src_block_dims[0] = srcDims[0];
+
+    auto orderAndReshape = [&](const size_t idx1, const size_t idx2, const size_t shift, const SizeVector& dims) {
+        for (size_t i = 0; i < nSpatialDims; i++) {
+            order[i * 2 + shift] = i + idx1;
+            order[i * 2 + shift + 1] = i + idx2;
+
+            optimizedParams.src_block_dims[order[i * 2 + shift]] = dims[i + shift];
+            optimizedParams.src_block_dims[order[i * 2 + shift + 1]] = blockSize;
+        }
+    };
+
+    if (isBlocked) {
+        SizeVector srcBlockedDims = getParentEdgeAt(0)->getDesc().getBlockingDesc().getBlockDims();
+        SizeVector dstBlockedDims = getChildEdgeAt(0)->getDesc().getBlockingDesc().getBlockDims();
+
+        size_t orderShiftForBlocks, orderShiftForDims;
+        if (mode == Mode::BLOCKS_FIRST) {
+            orderShiftForBlocks = 1;
+            orderShiftForDims = nSpatialDims + 2;
+
+            optimizedParams.src_block_dims[nSpatialDims + 1] = srcBlockedDims[1] / blockStep;
+            optimizedParams.src_block_dims[lastIdx] = srcBlockedDims.back();
+
+            order[1] = nSpatialDims + 1;
+            order[lastIdx] = lastIdx;
+        } else {
+            orderShiftForBlocks = nSpatialDims + 4;
+            orderShiftForDims = 3;
+
+            size_t newBlockSize = srcBlockedDims.back() / blockStep;
+            size_t newBlocksCount = srcBlockedDims[1] * newBlockSize / srcBlockedDims.back();
+            optimizedParams.src_block_dims[1] = newBlocksCount;
+            optimizedParams.src_block_dims[2] = srcBlockedDims[1] / newBlocksCount;
+            optimizedParams.src_block_dims[lastIdx - nSpatialDims] = newBlockSize;
+
+            order[1] = 1;
+            order[2] = 3;
+            order[lastIdx - 1] = 2;
+            order[lastIdx] = lastIdx - nSpatialDims;
+        }
+
+        orderAndReshape(orderShiftForDims, orderShiftForBlocks, 2, srcBlockedDims);
```

Comment:
firstSpatialOrder = 2 and share with planar

Reply:
Done, thanks